viziphant/statistics.py

```diff
@@ -41,3 +41,72 @@ def plot_isi(intervals, label, binsize=2*pq.ms, cutoff=250*pq.ms):
 
     return fig, ax
 
+
+def plot_patterns_statistics(patterns, winlen, binsize, n_neurons):
+    """
+    This function creates a histogram plot to visualise patterns statistics
+    output of a SPADE analysis.
+
+    Parameters
+    ----------
+    patterns : list
+        The output of elephant.spade
+    winlen : int
+        window length of the SPADE analysis
+    binsize : pq.Quantity
+        The bin size of the SPADE analysis
+    n_neurons : int
+        Number of neurons in the data set being analyzed
+
+    Returns
+    -------
+    fig : matplotlib.figure.Figure
+    ax : matplotlib.axes.Axes
+    """
+    patterns_dict = {'neurons': np.array([]),
+                     'occurrences': np.array([]),
+                     'pattern_size': np.array([]),
+                     'lags': np.array([])}
+    for pattern in patterns:
+        patterns_dict['neurons'] = \
+            np.append(patterns_dict['neurons'], pattern['neurons'])
+        patterns_dict['occurrences'] = \
+            np.append(patterns_dict['occurrences'], len(pattern['times']))
+        patterns_dict['pattern_size'] = \
+            np.append(patterns_dict['pattern_size'], len(pattern['neurons']))
+        patterns_dict['lags'] = \
+            np.append(patterns_dict['lags'], pattern['lags'])
+    if winlen == 1:
+        # case of only synchronous patterns
+        fig, axes = plt.subplots(3, 1, figsize=(10, 8))
+    else:
+        # case of patterns with delays
+        fig, axes = plt.subplots(4, 1, figsize=(10, 10))
+    plt.subplots_adjust(hspace=0.5)
+    axes[0].set_title('Patterns statistics')
+    axes[0].hist(patterns_dict['neurons'],
+                 bins=np.arange(0, n_neurons + 1))
```

Comment:
Doesn't the output of spade contain information about the number of neurons used in the analysis?

Reply:
Not really, unless all neurons participate in at least one pattern!